### PR TITLE
feat(project): updating angular from 12 to 13

### PR DESCRIPTION
**Pull request by shinyenyi**
- Migrated angular from version 12 to 13
- Migrated eslint from version 12 to 13
- Added variables: 
    const microphone = 'microphone' as PermissionName;
    const camera = 'camera' as PermissionName;
on files:
    commudle-ng/projects/shared-modules/hms-video/components/hms-video/conference/conference-settings/conference- 
    settings.component.ts
    commudle-ng/projects/shared-modules/hms-video/components/hms-video/local-preview/local-preview.component.ts
as 'microphone' and 'camera' are no longer part of the PermissionName types in typescript 4.4.

### CODE REVIEW

File: projects/commudle-admin/src/app/feature-modules/community-control-panel/community-control-panel.module.ts [hunk 1]
```diff
@@ -81,10 +80,5 @@ import { CommunityTeamComponent } from './components/community-team/community-te
     NbContextMenuModule,
     NbCheckboxModule,
   ],
-  entryComponents: [
-    CommunityFormsListStatsComponent,
-    CommunityFormsListActionsComponent,
-    // EmailerComponent
-  ],
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/community-control-panel/community-control-panel.module.ts:88]: Why were these removed?

File: projects/commudle-admin/src/polyfills.ts [hunk 2]
```diff
@@ -18,18 +18,6 @@
...
- */
-// import 'classlist.js';  // Run `npm install --save classlist.js`.
-
-/**
- * Web Animations `@angular/platform-browser/animations`
- * Only required if AnimationBuilder is used within the application and using IE/Edge or Safari.
- * Standard animation support in Angular DOES NOT require any polyfills (as of Angular 6.0).
- */
-// import 'web-animations-js';  // Run `npm install --save web-animations-js`.
-
```
Comment (arpansac) [inline projects/commudle-admin/src/polyfills.ts:32]: Why did we remove this?

File: .gitignore [hunk 3]
```diff
@@ -35,6 +35,7 @@ speed-measure-plugin*.json
 .history/*
 
 # misc
+/.angular/cache
```
Comment (arpansac) [inline .gitignore:38]: What's the purpose of this?

File: projects/shared-modules/hms-video/components/hms-video/conference/conference-settings/conference-settings.component.ts [hunk 4]
```diff
@@ -41,10 +41,13 @@ export class ConferenceSettingsComponent implements OnInit, OnDestroy {
...
   getMediaPermissions(): void {
+    const microphone = 'microphone' as PermissionName;
+    const camera = 'camera' as PermissionName;
+
     this.subscriptions.push(
       combineLatest(
-        this.localMediaService.getMediaPermissions('camera'),
-        this.localMediaService.getMediaPermissions('microphone'),
+        this.localMediaService.getMediaPermissions(camera),
+        this.localMediaService.getMediaPermissions(microphone),
```
Comment (arpansac) [inline projects/shared-modules/hms-video/components/hms-video/conference/conference-settings/conference-settings.component.ts:50]: @ArshdeepGrover please ensure this works perfectly

File: projects/lib-error-handler/tsconfig.lib.prod.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": "./tsconfig.lib.json",
   "angularCompilerOptions": {
-    "enableIvy": false
+    "compilationMode": "partial"
```
Comment (arpansac) [inline projects/lib-error-handler/tsconfig.lib.prod.json:4]: @Ashwin-op can you please confirm if this is correctly done?
@shinyenyi why have we switched to partial and removed ivy?

File: projects/shared-modules/hms-video/components/hms-video/local-preview/local-preview.component.ts [hunk 6]
```diff
@@ -39,10 +39,13 @@ export class LocalPreviewComponent implements OnInit, OnDestroy {
...
   getMediaPermissions(): void {
+    const microphone = 'microphone' as PermissionName;
+    const camera = 'camera' as PermissionName;
+
     this.subscriptions.push(
       combineLatest(
-        this.localMediaService.getMediaPermissions('camera'),
-        this.localMediaService.getMediaPermissions('microphone'),
+        this.localMediaService.getMediaPermissions(camera),
+        this.localMediaService.getMediaPermissions(microphone),
```
Comment (arpansac) [inline projects/shared-modules/hms-video/components/hms-video/local-preview/local-preview.component.ts:48]: Please check this one too @ArshdeepGrover

**Review by arpansac — CHANGES_REQUESTED** [review]
Great job @shinyenyi 
Just a few questions asked/changes suggested.

File: projects/commudle-admin/src/app/feature-modules/community-control-panel/community-control-panel.module.ts (hunk 1 above)
Comment (shinyenyi) [reply projects/commudle-admin/src/app/feature-modules/community-control-panel/community-control-panel.module.ts:88]: - EntryComponents used to create NgFactories in order to render components were only necessary till version 8 of angular. - In the migration to 13, they were removed by default because the Ivy renderer makes the need to specify components in the entryComponents array unnecessary and stores all the information required to render a component.

File: projects/commudle-admin/src/polyfills.ts (hunk 2 above)
Comment (shinyenyi) [reply projects/commudle-admin/src/polyfills.ts:32]: - Removed by default during migration because Internet Explorer 11 is not supported by Angular 13

File: .gitignore (hunk 3 above)
Comment (shinyenyi) [reply .gitignore:38]: From version 13.0.0 there will be a .angular folder generated in the root which is ignored by git containing a cache folder that caches builds.

File: projects/lib-error-handler/tsconfig.lib.prod.json (hunk 5 above)
Comment (shinyenyi) [reply projects/lib-error-handler/tsconfig.lib.prod.json:4]: this is recommended as angular 13 is 100% Ivy

File: projects/shared-modules/hms-video/components/hms-video/conference/conference-settings/conference-settings.component.ts (hunk 4 above)
Comment (ArshdeepGrover) [reply projects/shared-modules/hms-video/components/hms-video/conference/conference-settings/conference-settings.component.ts:50]: Working proper!!

File: projects/shared-modules/hms-video/components/hms-video/local-preview/local-preview.component.ts (hunk 6 above)
Comment (ArshdeepGrover) [reply projects/shared-modules/hms-video/components/hms-video/local-preview/local-preview.component.ts:48]: Works perfectly!
